include/itkBoneMorphometryFeaturesImageFilter.hxx

```diff
@@ -123,19 +123,21 @@ BoneMorphometryFeaturesImageFilter<TInputImage, TOutputImage, TMaskImage>
         // Iteration over the all neighborhood region
         for(NeighborIndexType nb = 0; nb<inputNIt.Size(); ++nb)
         {
-            tempOffset = inputNIt.GetOffset(nb);
-            if( !this->IsInsideNeighborhood(tempOffset) )
+            IndexType ind = inputNIt.GetIndex(nb);
+
+            if (!(this->IsInsideMaskRegion(ind, maskPointer->GetLargestPossibleRegion().GetSize())))
```

Comment:
Issues if `maskPointer` is null?

It would be better to use `maskPointer->GetBufferedRegion()` instead of `maskPointer->GetLargestPossibleRegion()` in case the entire mask has not been populated.